src/utilities/EventEmitter.test.ts

```diff
@@ -0,0 +1,250 @@
+import { it, describe, expect, beforeEach, afterEach, beforeAll, vi } from 'vitest';
+import EventEmitter, { removeListener } from './EventEmitter';
+
+describe('Removes a listener from an array or object of listeners', () => {
+  const listener = () => true;
+  const altListener = () => false;
+  
+  const targetListenerArrayFixture1 = [altListener, listener, altListener];
+  const targetListenerArrayFixture2 = [listener, altListener, listener];
+
+  it('Successfully removes a listener from arrays', () => {
+    const targetListeners = [[...targetListenerArrayFixture1], [...targetListenerArrayFixture2]];
```

Comment:
You could probably make a helper function for this since it is used in every test, or even put it into a `before`

Reply:
I think `targetListeners` is different in every test, do you think I should change the name? Or is it about a helper function spreading the array so it doesn't overwrite the source?

Reply:
Ah, if it's different in each one then I don't think we need a helper function.